roles/osp_observability/templates/prometheus.yml.j2

```diff
@@ -1,18 +1,34 @@
-#jinja2: trim_blocks:False
 global:
   scrape_interval: {{ prometheus_scrape_interval }}
   external_labels:
-    {% if ansible_fqdn %}agent_host: {{ ansible_fqdn }}{% endif %}
-    {% if openstack_cloud_name %}cloud_name: {{ openstack_cloud_name }}{% endif %}
+    {% if ansible_fqdn -%}
+    agent_host: {{ ansible_fqdn }}
+    {% endif -%}
+    {% if openstack_cloud_name -%}
+    cloud_name: {{ openstack_cloud_name }}
+    {% endif %}
 
 scrape_configs:
-{%- for service, nodes in scrape_endpoints.items() %}
+  {% for service, nodes in scrape_endpoints.items() -%}
   - job_name: {{ service }}
     static_configs:
       - targets: {{ nodes }}
-{%- endfor %}
+  {% endfor %}
 
 remote_write:
-{% for url in remote_write -%}
-  - url: {{ url }}
-{% endfor %}
+  {% for name, endpoint in remote_write.items() -%}
+  - url: {{ endpoint.url }}
+    {% if endpoint.basic_user is defined and endpoint.basic_pass is defined -%}
```

Comment:
It's getting a bit verbose, but you could also check if defined and length > 0 to really make sure you have a value here and not just an empty parameter set.

Reply:
https://github.com/infrawatch/osp-observability-ansible/pull/19/files#diff-7027c90a62527124195b5351e33095a6077349eaac7e013201e80bb567d2dcb8R7

Oh, you are quite right, actually. In that other PR I set defaults for these, and since there is no way to undefine them, the user would need to set them deliberately to `""`. I should change this to a length check.